CONTRIBUTING.md

```diff
@@ -41,35 +177,208 @@ proposed change so that the NIXL team can provide feedback.
+
+#### Why Our Review Process is Thorough
+
+- NIXL is used in performance-critical environments where reliability is paramount
+- We maintain backward compatibility and stable APIs for our users
+- Security and correctness are non-negotiable in data movement operations
+- We aim for code that is maintainable by the community long-term
+
+#### How We Support Contributors
 
-- The code style convention is enforced by common formatting tools
-  for a given language (such as clang-format for c++, black for python).
-  See below on how to ensure your contributions conform. In general please follow
-  the existing conventions in the relevant file, submodule, module,
-  and project when you add new code or when you extend/fix existing
-  functionality.
+- Reviewers provide detailed feedback to help improve the contribution
+- We're here to collaborate, not just critique - ask questions if feedback is unclear
+- For significant changes, we may suggest incremental PRs to make review easier
+- The team will help ensure your contribution aligns with NIXL's architecture
 
-- Avoid introducing unnecessary complexity into existing code so that
-  maintainability and readability are preserved.
+#### Tips for Smoother Reviews
 
-- Try to keep code changes for each pull request (PR) as concise as possible:
+- Start with smaller PRs to familiarize yourself with our standards
+- Engage early through issues for design discussions
+- Be responsive to feedback and ask for clarification when needed
+- Consider breaking large changes into logical, reviewable chunks
 
-  - Fillout PR template with clear description and mark applicable checkboxes
+We recognize that our standards are high, but this ensures NIXL remains a reliable foundation for critical workloads. Your patience and collaboration through the review process is genuinely appreciated, and we're committed to helping your contribution succeed.
 
-  - Avoid committing commented-out code.
+## Plugin Development
 
-  - Wherever possible, each PR should address a single concern. If
-    there are several otherwise-unrelated things that should be fixed
-    to reach a desired endpoint, it is perfectly fine to open several
-    PRs and state in the description which PR depends on another
-    PR. The more complex the changes are in a single PR, the more time
-    it will take to review those changes.
+### Creating New Plugins
 
-  - Make sure that the build log is clean, meaning no warnings or
-    errors should be present.
+When contributing new plugins, follow these guidelines:
 
-  - Make sure all tests pass.
+#### 1. Plugin Structure
+
+Plugins are located in `src/plugins/`. Your plugin should follow this structure:
+
+```text
+src/plugins/your_plugin/
+├── meson.build
+├── your_plugin.cpp
+├── your_plugin.h
+├── your_backend.cpp
+├── your_backend.h
+└── README.md
+```
+
+Tests should be added in the GoogleTest-based test directory:
+
+```text
+test/gtest/unit/plugins/your_plugin/
```

Comment:
Should we change the way that users write unit tests? - add something to guide them there?
Do we have information about adding unit tests to CI build?